.github/workflows/on_call_do_publish_image.yml

```diff
@@ -0,0 +1,115 @@
+name: CI Release Images
+
+on:
+  workflow_call:
+    inputs:
+      repository:
+        description: 'Name of the repo to download server artifact from'
+        type: string
+      run-id:
+        description: 'Id of the run to download server artifact from'
+        type: string
+      art-name:
+         description: 'Artifact name to download'
+         type: string
+      tags:
+        description: 'Tags for the image'
+        type: string
+      push:
+         description: 'If true push image.'
+         default: true
+         type: boolean
+
+      branch:
+         description: 'infinispan-image branch to checkout.'
+         default: main
+         type: string
+    secrets:
+      token:
+         required: true
+         description: 'GH token'
+      quay_user:
+         required: true
+      quay_pass:
+         required: true
+  workflow_dispatch:
+    inputs:
+      repository:
+        description: 'Name of the repo to download server artifact from'
+        type: string
+      run-id:
+        description: 'Id of the run to download server artifact from'
+        type: string
+      art-name:
+        description: 'Artifact name to download'
+        type: string
+      tags:
+       description: 'Tags for the image'
+       type: string
+      push:
+         description: 'If true push image.'
+         default: true
+         type: boolean
+      branch:
+         description: 'infinispan-image branch to checkout.'
+         default: main
+         type: string
+
+jobs:
+   image:
+      runs-on: ubuntu-latest
+      steps:
+         - name: Checkout
+           uses: actions/checkout@v4
+           with:
+               repository: 'infinispan/infinispan-images'
+               ref: ${{ inputs.branch }}
+
+         - name: Download Artifact
+           uses: actions/download-artifact@v4.1.7
+           with:
+             repository: '${{ inputs.repository }}'
+             run-id: '${{ inputs.run-id }}'
+             name:  '${{ inputs.art-name }}'
+             github-token: ${{ secrets.token }}
+
+         - name: Unzip Artifact
+           id: unzip_artifact
+           run: |
+               echo "serverzip=$(ls infinispan-server-*.zip)" >> $GITHUB_OUTPUT
+         - name: Install CEKit
+           uses: cekit/actions-setup-cekit@v1.1.5
+
+         - name: Create Dockerfile
+           env:
+             SERVER_ZIP: ${{ steps.unzip_artifact.outputs.serverzip }}
+             TAGS: ${{ inputs.tags }}
+           run: |
+               pwd
+               ls
+               SERVER_OVERRIDE="{\"artifacts\":[{\"name\":\"server\",\"path\":\"${GITHUB_WORKSPACE}/${SERVER_ZIP}\"}]}"
+               TAG="${TAGS%%,*}"
+               IMG_NAME="${TAG%%:*}"
+               IMG_VER="${TAG##*:}"
+               cekit -v --descriptor server-openjdk.yaml build --overrides '{"version": "'${IMG_VER}'", "name": "'${IMG_NAME}'"}' --overrides ${SERVER_OVERRIDE} --dry-run docker
+
+         - name: Set up Docker Buildx
```

Comment:
I think the flow of the action is clearer as a reader if we move all of the dependency and initialization steps to the start of the job, followed by the actual logic of the action. A benefit of this, is that if any of the required dependencies/login steps fail, we're failing-fast and no steps are executed redundantly.